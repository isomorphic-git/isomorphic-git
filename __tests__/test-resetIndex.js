/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
const snapshots = require('./__snapshots__/test-resetIndex.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
const { resetIndex, listFiles } = require('isomorphic-git')

describe('resetIndex', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('modified', async () => {
    // Setup
    let { fs, gitdir, dir } = await makeFixture('test-resetIndex')
    // Test
    let before = await listFiles({ fs, gitdir })
    expect(before).toMatchSnapshot()
    await resetIndex({ fs, dir, gitdir, filepath: 'a.txt' })
    let after = await listFiles({ fs, gitdir })
    expect(after).toMatchSnapshot()
    expect(before.length === after.length).toBe(true)
  })
  it('new', async () => {
    // Setup
    let { fs, gitdir, dir } = await makeFixture('test-resetIndex')
    // Test
    let before = await listFiles({ fs, gitdir })
    expect(before).toMatchSnapshot()
    await resetIndex({ fs, dir, gitdir, filepath: 'd.txt' })
    let after = await listFiles({ fs, gitdir })
    expect(after).toMatchSnapshot()
    expect(before.length === after.length + 1).toBe(true)
  })
})
