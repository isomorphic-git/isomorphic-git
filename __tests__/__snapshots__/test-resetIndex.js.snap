// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resetIndex modified 1`] = `
Array [
  "a.txt",
  "b.txt",
  "d.txt",
]
`;

exports[`resetIndex modified 2`] = `
Array [
  "a.txt",
  "b.txt",
  "d.txt",
]
`;

exports[`resetIndex new 1`] = `
Array [
  "a.txt",
  "b.txt",
  "d.txt",
]
`;

exports[`resetIndex new 2`] = `
Array [
  "a.txt",
  "b.txt",
]
`;
